Tests/KeyboardNavigatorTests.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  KeyboardNavigatorTests.swift
+//  KeyboardSupport-iOSTests
+//
+//  Created by Earl Gaspard on 8/3/18.
+//  Copyright © 2018 Bottle Rocket. All rights reserved.
+//
+
+import XCTest
+@testable import KeyboardSupport
+
+class KeyboardNavigatorTests: XCTestCase {
+
+    // MARK: - Properties
+    
+    private let keyboardToolbar = KeyboardToolbar()
+    
+    // MARK: - Tests
+    
+    func test_KeyboardNavigator_InitializesWithTextFields() {
+        let keyboardNavigator = keyboardNavigatorWithTextFields()
+        
+        XCTAssertEqual(keyboardNavigator.textInputs.count, 3)
+        XCTAssertEqual(keyboardNavigator.keyboardToolbar, keyboardToolbar)
+        XCTAssertTrue(keyboardNavigator.returnKeyNavigationEnabled)
+        XCTAssertEqual(keyboardNavigator.currentTextInputIndex, 0)
+    }
+    
+    func test_KeyboardNavigator_InitializesWithTextViews() {
+        let keyboardNavigator = keyboardNavigatorWithTextViews()
+        
+        XCTAssertEqual(keyboardNavigator.textInputs.count, 3)
+        XCTAssertEqual(keyboardNavigator.keyboardToolbar, keyboardToolbar)
+        XCTAssertTrue(keyboardNavigator.returnKeyNavigationEnabled)
+        XCTAssertEqual(keyboardNavigator.currentTextInputIndex, 0)
+    }
+    
+    func test_KeyboardNavigator_AccessoryDidTapBack() {
+        let textFieldNavigator = keyboardNavigatorWithTextFields()
+        textFieldNavigator.keyboardAccessoryDidTapNext(UIView())
+        textFieldNavigator.keyboardAccessoryDidTapBack(UIView())
+        
+        let textViewNavigator = keyboardNavigatorWithTextViews()
+        textViewNavigator.keyboardAccessoryDidTapNext(UIView())
+        textViewNavigator.keyboardAccessoryDidTapBack(UIView())
+    }
+    
+    func test_KeyboardNavigator_AccessoryDidTapNext() {
+        let textFieldNavigator = keyboardNavigatorWithTextFields()
+        textFieldNavigator.keyboardAccessoryDidTapNext(UIView())
+        
+        let textViewNavigator = keyboardNavigatorWithTextViews()
+        textViewNavigator.keyboardAccessoryDidTapNext(UIView())
+    }
+    
+    func test_KeyboardNavigator_AccessoryDidTapDone() {
+        let textFieldNavigator = keyboardNavigatorWithTextFields()
+        textFieldNavigator.keyboardAccessoryDidTapDone(UIView())
+        
+        let textViewNavigator = keyboardNavigatorWithTextViews()
+        textViewNavigator.keyboardAccessoryDidTapDone(UIView())
+    }
+}
+
+extension KeyboardNavigatorTests: KeyboardNavigatorDelegate {
```

Comment:
The way these tests are implemented on the `KeyboardNavigatorDelegate` doesn't feel quite right to me. For example, "tapping keyboard accessory next button asserts that the `keyboardNavigatorDidTapBack` delegate callback was invoked". I think the better way to to do this would be to create a mock that acts as the delegate and then we check properties on the mock to make sure things were called as expected.